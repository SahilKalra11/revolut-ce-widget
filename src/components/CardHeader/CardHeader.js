import React from 'react';
import './CardHeader.scss';
import {getCurrencySymbol} from '../../helpers';

const CardHeader = ({rates, currencyTo, currencyFrom, onExchange, onCancel, valueFrom}) => {
    let course = `${getCurrencySymbol(currencyFrom)}1 = ${getCurrencySymbol(currencyTo)}${rates[currencyTo]}`;
    const isDisabled = (currencyTo === currencyFrom) || !valueFrom;
    return (
        <header className="card-header">
            <button
                disabled={isDisabled}
                type="button"
                className="btn"
                onClick={onCancel}>Cancel
            </button>
            <div className="card-header_rate">{course}</div>
            <button
                disabled={isDisabled}
                type="submit"
                className="btn"
                onClick={onExchange}>Exchange
            </button>
        </header>
    )
};
export default CardHeader;
