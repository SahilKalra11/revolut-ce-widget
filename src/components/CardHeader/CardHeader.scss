@import '../../utilities/variables';

.card-header {
  color: $color-white;
  display: flex;
  justify-content: space-between;
  padding: $padding/2 $padding;
  

  &__list {
    list-style: none;
    display: flex;
  }
}
