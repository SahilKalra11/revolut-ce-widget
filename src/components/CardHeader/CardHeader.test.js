import React from 'react';
import ReactDOM from 'react-dom';
import render from 'react-test-renderer';
import CardHeader from './CardHeader';
import {unmountComponentAtNode} from "react-dom";

describe('CardHeader', () => {
    let container = null;
    beforeEach(() => {
        container = document.createElement("div");
        document.body.appendChild(container);
    });

    afterEach(() => {
        unmountComponentAtNode(container);
        container.remove();
        container = null;
    });

    const props = {
        currencyFrom: 'USD',
        currencyTo: 'EUR',
        valueTo: 10,
        rates: {
            USD: 1,
            EUR: 1.20,
            GBP: 1.3
        },
        onExchange: jest.fn(),
        onCancel: jest.fn()
    };

    it('renders component without crashing', () => {
        ReactDOM.render(
            <CardHeader {...props} />,
            container
        );
    });

    test('matches to snapshot', () => {
        const component = render.create(
            <CardHeader {...props} />
        );
        let componentTree = component.toJSON();
        expect(componentTree).toMatchSnapshot();
    });
});
