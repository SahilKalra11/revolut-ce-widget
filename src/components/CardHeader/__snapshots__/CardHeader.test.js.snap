// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardHeader matches to snapshot 1`] = `
<header
  className="card-header"
>
  <button
    className="btn"
    disabled={true}
    onClick={[MockFunction]}
    type="button"
  >
    Cancel
  </button>
  <div
    className="card-header_rate"
  >
    $1 = â‚¬1.2
  </div>
  <button
    className="btn"
    disabled={true}
    onClick={[MockFunction]}
    type="submit"
  >
    Exchange
  </button>
</header>
`;
